source/MRMesh/MROffset.cpp

```diff
@@ -46,19 +46,28 @@ tl::expected<Mesh, std::string> offsetMesh( const MeshPart & mp, float offset, c
 
     auto voxelSizeVector = Vector3f::diagonal( voxelSize );
     // Make grid
-    auto grid = ( !useShell ) ?
-        // Make level set grid if it is closed
-        meshToLevelSet( mp, AffineXf3f(), voxelSizeVector, std::abs( offsetInVoxels ) + 2,
-            subprogress( params.callBack, 0.0f, signPostprocess ? 0.33f : 0.5f ) ) :
-        // Make distance field grid if it is not closed
-        meshToDistanceField( mp, AffineXf3f(), voxelSizeVector, std::abs( offsetInVoxels ) + 2,
+    FloatGrid grid;
+    if ( !useShell && !signPostprocess ) 
+    {
+        // Compute signed distance grid
+        grid = meshToLevelSet( mp, AffineXf3f(), voxelSizeVector, std::abs( offsetInVoxels ) + 2,
+            subprogress( params.callBack, 0.0f, signPostprocess ? 0.33f : 0.5f ) );
```

Comment:
in this block we know that `!signPostprocess` so `subprogress( params.callBack, 0.0f, 0.5f ) )`